cmd/internal/cli/build.go

```diff
@@ -285,6 +287,22 @@ var buildRemoteFlag = cmdline.Flag{
 	Hidden:       true,
 }
 
+var buildVarArgsFlag = cmdline.Flag{
+	ID:           "buildVarArgsFlag",
+	Value:        &buildArgs.buildVarArgs,
+	DefaultValue: []string{},
+	Name:         "build-arg",
+	Usage:        "--build-arg is used for passing variables to build procedure",
+}
+
+var buildVarArgFileFlag = cmdline.Flag{
+	ID:           "buildVarArgFileFlag",
+	Value:        &buildArgs.buildVarArgFile,
+	DefaultValue: "",
+	Name:         "build-arg-file",
+	Usage:        "--build-arg-file is used for passing variables file to build procedure",
```

Comment:
Use "specifies a file containing variable=value lines to replace '{{variable}}' with value in build definition files"

Reply:
thanks! resolved in the latest commit.
https://github.com/apptainer/apptainer/pull/1344/commits/1842bc374a7db1c7a72fb8fa11d811d7de4e2ff4